x/crosschain/keeper/msg_server_migrate_erc20_custody_funds.go

```diff
@@ -0,0 +1,104 @@
+package keeper
+
+import (
+	"context"
+
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	authoritytypes "github.com/zeta-chain/zetacore/x/authority/types"
+	"github.com/zeta-chain/zetacore/x/crosschain/types"
+)
+
+// MigrateERC20CustodyFunds migrates the funds from the current TSS to the new TSS
+func (k msgServer) MigrateERC20CustodyFunds(
+	goCtx context.Context,
+	msg *types.MsgMigrateERC20CustodyFunds,
+) (*types.MsgMigrateERC20CustodyFundsResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// check if authorized
+	err := k.GetAuthorityKeeper().CheckAuthorization(ctx, msg)
+	if err != nil {
+		return nil, errorsmod.Wrap(authoritytypes.ErrUnauthorized, err.Error())
+	}
+
+	// get the current TSS nonce allow to set a unique index for the CCTX
```

Comment:
Consider adding a check for PenidingNonces, such that there are no pending cctx when the migration is done

Reply:
I don't think this is necessary to block all cctx when migrating ERC20Custody fund.
We will:
- pause the ERC20Custody to block deposit
- pause the associated ZRC20